### PR TITLE
Fixed issue flash size on Handwired K552 keyboard

### PR DESCRIPTION
**Pull request by HorrorTroll**
## Description

It was mistake to use ldscript from STM32F103x8, instead using STM32F103xC. So the flash size was limited at 64kb, while it should be 256kb

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by drashna — APPROVED** [review]

**Comment by tzarc** [comment]
As far as I can tell, you can do all of this in a much simpler way.

* Remove the custom board files (i.e. `boards/STM32_F103XC_STM32DUINO` in its entirety)
* Swap board to the normal STM32duino: `BOARD = STM32_F103_STM32DUINO`
* Remove the other customisations in the `rules.mk` apart from the `MCU_LDSCRIPT`
* Add a keyboard-level `board.h` file with the content:
```c
#pragma once
#include_next <board.h>
#undef STM32F103xB
#define STM32F103xE
```
* Add a validation inside `handwired_k552.c`:
```c
#ifndef STM32F103xE
#error "Should be building for F103xE but wasn't!"
#endif
```

**Comment by tzarc** [comment]
Ignore the change to target develop; I saw board files and automatically assumed it was a core modification.

File: keyboards/horrortroll/handwired_k552/rules.mk [hunk 1]
```diff
@@ -1,11 +1,13 @@
...
-
-# Bootloader selection
-BOOTLOADER = stm32duino
+# Custom loader configuration
+MCU_LDSCRIPT = STM32F103xC_stm32duino_bootloader
+OPT_DEFS += -DBOOTLOADER_STM32DUINO
+BOARD = STM32_F103XC_STM32DUINO
+STM32_BOOTLOADER_ADDRESS = 0x80000000
+DFU_ARGS = -d 1EAF:0003 -a 2 -R
+DFU_SUFFIX_ARGS = -v 1EAF -p 0003
```
Comment (fauxpark) [inline keyboards/horrortroll/handwired_k552/rules.mk:10]:
```suggestion
# Custom loader configuration
BOOTLOADER_TYPE = stm32duino
OPT_DEFS += -DBOOTLOADER_STM32DUINO
MCU_LDSCRIPT = STM32F103xC_stm32duino_bootloader
BOARD = STM32_F103XC_STM32DUINO
DFU_ARGS = -d 1EAF:0003 -a 2 -R
DFU_SUFFIX_ARGS = -v 1EAF -p 0003
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/horrortroll/handwired_k552/rules.mk (hunk 1 above)
Comment (HorrorTroll) [reply keyboards/horrortroll/handwired_k552/rules.mk]: fixed now, thanks

**Comment by HorrorTroll** [comment]
Just pushed an updated code, testing and working normally

**Comment by HorrorTroll** [comment]
Closed

**Pull request closed without merging**